src/mds/Beacon.cc

```diff
@@ -396,7 +398,10 @@ void Beacon::notify_health(MDSRank const *mds)
         dout(20) << "Session servicing RECALL " << session->info.inst
           << ": " << session->recalled_at << " " << session->recall_release_count
           << "/" << session->recall_count << dendl;
-        if (session->recalled_at < cutoff) {
+	if (last_recall < cutoff || session->recalled_at < last_recall) {
+	  dout(20) << "  no longer recall" << dendl;
+	  session->clear_recalled_at();
```

Comment:
Hmm, so with the new behaviour we will not give a health warning in the case where one client is bad, but another client is trimming enough cache entries to get the overall MDS total back to within limits.  I think that's okay though, I don't see a better way.